app/controllers/screenings_controller.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class ScreeningsController < ApplicationController
+  def index
+    @screenings = Screening.all.map do |screening|
+      screening_hash(screening)
+    end
+    render json: @screenings
+  end
+
+  def show
+    screening = Screening.find(params[:id])
+    transformed_screening = screening_hash(screening)
+    render json: transformed_screening
+  end
+
+  def update
+    screening = Screening.find(params[:id])
+    if screening.update(screening_params)
+      render json: @screening
+    else
+      render json: @screening.errors, status: :unprocessable_entity
+    end
+  end
+
+  def create
+    @screening = Screening.new(screening_params)
+
+    if @screening.save
+      generate_seat(@screening)
```

Comment:
What will happen if you will be able to create Screening, but in the middle of generating seats process something fails? You will have probably inconsistency on the database level.
You've had already training about DB transactions, consider if it's not a good place to use this knowledge :)

https://api.rubyonrails.org/v6.1.4/classes/ActiveRecord/Transactions/ClassMethods.html